semeio/workflows/localisation/local_script_lib.py

```diff
@@ -0,0 +1,444 @@
+                kw_config_model = node.getKeywordModelConfig()
+                my_param.parameters = kw_config_model.getKeyWords().strings
+            elif impl_type in implementation_type_not_scalar:
+                # Node contains parameter from FIELD, SURFACE or GEN_PARAM
+                # The parameters_for_node dict contains empty list for FIELD
+                # and SURFACE.
+                # The number of variables for a FIELD parameter is
+                # defined by the grid in the GRID keyword.
+                # The number of variables for a SURFACE parameter is
+                # defined by the size of a surface object.
+                # For GEN_PARAM the list contains the number of variables.
+                # parameters_for_node[key] = []
+                if impl_type == ErtImplType.GEN_DATA:
+                    gen_data_config = node.getDataModelConfig()
+                    data_size = gen_data_config.get_initial_size()
+                    if data_size <= 0:
+                        print(
+                            f"\n -- Warning: A GEN_PARAM node {key} "
+                            "is defined in ERT config.\n"
+                            "          This requires that the localization script "
+                            "must be run AFTER \n"
+                            "          initialization of the ensemble, "
+                            "but BEFORE the first update step.\n"
+                        )
+                    debug_print(
+                        f"GEN_PARAM node {key} has {data_size} parameters.",
+                        LogLevel.LEVEL3,
+                    )
+                    my_param.parameters = [str(item) for item in range(data_size)]
+
+    return new_params
+
+
+def read_localisation_config(args):
+    if len(args) == 1:
+        specification_file_name = args[0]
+    else:
+        raise ValueError(f"Expecting a single argument. Got {args} arguments.")
+
+    print(
+        "\n" f"Define localisation setup using config file: {specification_file_name}"
+    )
+    with open(specification_file_name, "r") as yml_file:
+        localisation_yml = yaml.safe_load(yml_file)
+    return localisation_yml
+
+
+def active_index_for_parameter(node_name, param_name, ert_param_dict):
+    # For parameters defined as scalar parameters (coming from GEN_KW)
+    # the parameters for a node have a name. Get the index from the order
+    # of these parameters.
+    # For parameters that are not coming from GEN_KW,
+    # the active index is not used here.
+
+    ert_param_list = ert_param_dict[node_name]
+    if len(ert_param_list) > 0:
+        index = -1
+        for count, name in enumerate(ert_param_list):
+            if name == param_name:
+                index = count
+                break
+        assert index > -1
+    else:
+        index = None
+    return index
+
+
+def activate_gen_kw_param(model_param_group, node_name, param_list, ert_param_dict):
```

Comment:
Should be possible to merge this with `activate_gen_param`?

Reply:
Since GEN_KW uses parameter names and GEN_PARAM uses indices only to identify parameters, I think it is OK to have two separate functions, otherwise we have to check on implementation type on this level also if we merge active_gen_param with activate_gen_kw_param